src/runtime/schema.ts

```diff
@@ -211,11 +225,15 @@ export class Schema {
       if (fields[name] == undefined) {
         return AtLeastAsSpecific.NO;
       }
-      if (type.kind && type.kind === 'schema-nested') {
-        if (!(fields[name].kind && fields[name].kind === 'schema-nested')) {
+      if (type.kind && ['schema-nested', 'schema-reference', 'schema-collection'].includes(type.kind)) {
```

Comment:
Should this list be pulled into the high level? I've been storing a set of them in the `manifest-ast-nodes.ts` file.

Ideally we'd extra the list of possible kinds from a type but I haven't worked out how to extract the kinds from a union type in Typescript yet.

They do have it for enums `Object.values(enumType)` gets the list of possible values. Any ideas @shans

Reply:
it would be a good change; we are going to refactor schema field types more comprehensively once we're not restricted by g3 compilation semantics so probably could happen as part of that.

Reply:
IIUC there is no immediate action item from this comment in this PR. I'll merge as is and address in follow up, as needed.
I'll try to create classes for the schema fields next week, we'll see how far I get with the no-cycles restriction though:)